licensing/src/main/java/io/helidon/build/licensing/Report.java

```diff
@@ -330,10 +363,169 @@ private boolean generateAttributionFile(AttributionDocument attributionDocument,
+        boolean first = true;
+        for (AttributionDependency d : deps) {
+            HashSet<String> intersection = new HashSet<>(moduleList);
+            intersection.retainAll(d.getConsumers());
+            if (moduleList.isEmpty() || !intersection.isEmpty()) {
+                if (first) {
+                    w.write("\t\"dependencies\": [\n");
+                    w.write("\t\t{\n");
+                    first = false;
+                } else {
+                    w.write("\t\t, {\n");
+                }
+
+                w.write(String.format("\t\t\t\"name\": \"%s\"\n", jsonEscape(d.getName())));
+                if (includeVersion)
+                    w.write(String.format("\t\t\t, \"version\": \"%s\"\n", jsonEscape(d.getVersion())));
+                w.write(String.format("\t\t\t, \"licensor\": \"%s\"\n", jsonEscape(d.getLicensor())));
+                w.write(String.format("\t\t\t, \"license-name\": \"%s\"\n", jsonEscape(d.getLicenseName())));
+                w.write(String.format("\t\t\t, \"attribution\": \"%s\"\n", jsonEscape(d.getAttribution())));
+                w.write(String.format("\t\t\t, \"used-by\": [\n"));
+                boolean firstConsumer = true;
+                for (String consumer : d.getConsumers()) {
+                    if (firstConsumer) {
+                        w.write(String.format("\t\t\t\t\"%s\"\n", jsonEscape(consumer)));
+                        firstConsumer = false;
+                    } else {
+                        w.write(String.format("\t\t\t\t, \"%s\"\n", jsonEscape(consumer)));
+                    }
+                }
+                w.write("\t\t\t]\n");
+                w.write("\t\t}\n");
+
+                detectLicenses(licensesUsed, d.getAttribution());
+            }
+        }
+        w.write("\t]\n");
+
+        // If we haven't written anything, then let the caller know
+        if (first) {
+            w.write("}\n");
+            return false;
+        }
+
+        // Write full text of licenses used (that were squashed out of report) to the
+        // end of the file.
+        first = true;
+        for (String s : licensesUsed) {
+            if (first) {
+                w.write("\t, \"licenses\" : {\n");
+            }
+
+            // Get license text for AttributionDocument
+            AttributionLicense license = getLicense(attributionDocument, s);
+            String sJson = jsonEscape(s);
+            if (license != null) {
+                w.write(String.format("\t\t%s\"%s\": \"%s\"\n", (first ? "":", "), sJson, jsonEscape(license.getText())));
+            } else {
+                w.write(String.format("\t\t%s\"%s\": \"\"No license text found for %s\"\n", (first ? "":", "), sJson, sJson));
+            }
+
+            first = false;
+        }
+        w.write("\t}\n");
+
+        w.write("}\n");
+
+        return true;
+    }
+
+    private static String jsonEscape(String str) {
+        return str
```

Comment:
Need a null check here. I ran into an NPE while testing with some of our license data. I added this to work-around:
```
if (str == null) {
    return "";
}
```